lib/Optimizer/Builder/Intrinsics.cpp

```diff
@@ -371,6 +371,136 @@ static constexpr IntrinsicCode intrinsicTable[] = {
     {cudaq::stdvecBoolCtorFromInitList, {}, R"#(
   func.func private @__nvqpp_initializer_list_to_vector_bool(!cc.ptr<none>, !cc.ptr<none>, i64) -> ())#"},
 
+    {"__nvqpp_internal_number_of_digits", {}, R"#(
+  func.func private @__nvqpp_internal_number_of_digits(%arg0: i64) -> i64 {
```

Comment:
Is this function only used for the integer->string conversion?

Reply:
Yeah, it's only used for the integer->string conversion.
Originally, I wanted to make the `to_string` produce a compact string. This string then needs to be assembled with prefix and postfix, thus having this as a separate function to help with the allocation of the integer string and the final string buffers.

Reply:
We've moved this into an (upcoming) separate PR.